src/scripts/main.js

```diff
@@ -1,3 +1,160 @@
 'use strict';
 
-// write code here
+const tableHeaderRow = document.querySelector('thead tr');
+const tbody = document.querySelector('tbody');
+const rows = Array.from(tbody.querySelectorAll('tr'));
+
+let isAscending = true;
+let lastSortedColumn = null;
+
+function sortElements(columnIndex) {
+  rows.sort((a, b) => {
+    let aValue = a.cells[columnIndex].textContent;
+    let bValue = b.cells[columnIndex].textContent;
+
+    if (columnIndex >= 3) {
+      aValue = parseFloat(aValue.replace(/[^0-9.-]+/g, ''));
+      bValue = parseFloat(bValue.replace(/[^0-9.-]+/g, ''));
+    }
+
+    if (aValue < bValue) {
+      return isAscending ? -1 : 1;
+    } else if (aValue > bValue) {
+      return isAscending ? 1 : -1;
+    } else {
+      return 0;
+    }
+  });
```

Comment:
The sorting logic assumes that columns with an index of 3 or higher contain numeric values. Ensure that this assumption is correct for your table. If there are non-numeric columns beyond index 2, the sorting might not work as expected.